composer/loggers/object_store_logger.py

```diff
@@ -42,21 +33,32 @@ def _always_log(state: State, log_level: LogLevel, artifact_name: str):
     return True
 
 
+def _build_object_store(object_store_cls: Type[ObjectStore], object_store_kwargs: Dict[str, Any]):
+    # error: Expected no arguments to "ObjectStore" constructor
+    return object_store_cls(**object_store_kwargs)  # type: ignore
+
+
 class ObjectStoreLogger(LoggerDestination):
     r"""Logger destination that uploads artifacts to an object store.
 
     This logger destination handles calls to :meth:`~composer.loggers.logger.Logger.file_artifact`
-    and uploads files to an object store, such as AWS S3 or Google Cloud Storage.
+    and uploads files to :class:`.ObjectStore`, such as AWS S3 or Google Cloud Storage.
 
     .. testcode:: composer.loggers.object_store_logger.ObjectStoreLogger.__init__
 
+        from composer.loggers import ObjectStoreLogger
+        from composer.utils import LibcloudObjectStore
+
         object_store_logger = ObjectStoreLogger(
-            provider='s3',
-            container='my-bucket',
-            provider_kwargs={
-                'key': 'AKIA...',
-                'secret': '*********',
-                'region': 'ap-northeast-1',
+            object_store_cls=LibcloudObjectStore,
+            object_store_kwargs={
```

Comment:
Could you provide some context as to why we have to accept the class and kwargs, rather than an instance of the class?

Reply:
This is to support multiple upload workers. A different instance object store is constructed within each worker. It is not safe to share an `ObjectStore` instance across thread boundaries, and it would be impossible to share object stores across subprocesses. I'll add a comment to clarify what's going on.